src/ui/App.tsx

```diff
@@ -0,0 +1,380 @@
+        }[];
+      } = await response.json();
+
+      if (data.status === 'error' && data.errors) {
+        setSubmitErrors(
+          data.errors.flatMap((error) => {
+            if (Array.isArray(error.message)) {
+              return error.message;
+            }
+            return error.message;
+          }),
+        );
+      }
+
+      if (data.status === 'success') {
+        setIsSuccess(true);
+      }
+    } catch (err) {
+      const errorMessage =
+        err instanceof Error ? err.message : 'An error occurred';
+      setError(errorMessage);
+      console.error('Error submitting selections:', err);
+      alert('Failed to submit selections. Please try again.');
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  const fetchChangelog = async (packageName: string): Promise<void> => {
+    setLoadingChangelogs((prev) => ({ ...prev, [packageName]: true }));
+    try {
+      const response = await fetch(`/api/changelog?package=${packageName}`);
+      if (!response.ok) {
+        throw new Error('Failed to fetch changelog');
+      }
+      const changelog = await response.text();
+      setChangelogs((prev) => ({ ...prev, [packageName]: changelog }));
+    } catch (err) {
+      const errorMessage =
+        err instanceof Error ? err.message : 'Failed to fetch changelog';
+      setError(errorMessage);
+    } finally {
+      setLoadingChangelogs((prev) => ({ ...prev, [packageName]: false }));
+    }
+  };
+
+  const handleBulkAction = (action: ReleaseType) => {
+    const newSelections = { ...selections };
+    selectedPackages.forEach((packageName) => {
+      newSelections[packageName] = action;
+    });
+    setSelections(newSelections);
+    setSelectedPackages(new Set());
+  };
+
+  const togglePackageSelection = (packageName: string) => {
+    setSelectedPackages((prev) => {
+      const newSet = new Set(prev);
+      if (newSet.has(packageName)) {
+        newSet.delete(packageName);
+      } else {
+        newSet.add(packageName);
+      }
+      return newSet;
+    });
+  };
+
+  if (isSubmitting) {
+    return (
+      <div className="fixed inset-0 bg-white bg-opacity-90 flex items-center justify-center z-50">
+        <div className="text-center p-8">
+          <h2 className="text-2xl font-bold mb-4">Processing Release</h2>
+          <p className="mb-6">
+            Please wait while we process your release. This may take a few
+            minutes...
+          </p>
+          <div className="animate-spin rounded-full h-12 w-12 border-b-2 border-blue-500 mx-auto"></div>
+        </div>
+      </div>
+    );
+  }
+
+  if (isSuccess) {
+    return (
+      <div className="fixed inset-0 bg-white flex items-center justify-center">
+        <div className="text-center p-8">
+          <h2 className="text-2xl font-bold mb-4">Release Successful!</h2>
+          <p className="mb-6">You can now close this window.</p>
+          <button
+            onClick={() => window.close()}
```

Comment:
This `window.close` doesn't seem to work. I get an error message in devtools: `Scripts may close only the windows that were opened by them.`

Maybe we don't need this button?